src/term/transform/scott_generation.rs

```diff
@@ -5,26 +5,22 @@ impl Book {
     for adt in self.adts.values() {
       let curr_ctrs = adt.ctrs.clone();
 
-      let mut ctr_args = Vec::new();
-      let mut ctrs = Vec::new();
-
       for (ctr_name, arity) in &curr_ctrs {
+        let mut ctr_args = Vec::new();
+        let mut ctrs = Vec::new();
+
         for ctr in curr_ctrs.keys() {
           ctrs.push(ctr.clone());
           if ctr == ctr_name {
             make_args(arity, &mut ctr_args);
```

Comment:
Why is this inside an `if` that will only match once? You can just move this line outside the `for`